enhancements/installer/signal-cluster-deletion.md

```diff
@@ -68,6 +68,8 @@ spec:
 
 Add an admission plugin that prevents delete when `blockTeardown` has been set.
 
+We believe it would be best to add the CRD and admission plugin to the cluster version operator. We don't foresee this requiring a controller and doing a separate operator seems like overkill for this object.
+
```

Comment:
@LalatenduMohanty Heads up regarding this enhancement. We think it would make the most sense to add this to the CVO. Please let me know if you have any questions or concerns.

Reply:
No, since CVO is instrumental for working of the cluster.

`hey there is no owner for this object, let put in CVO` should be immediately stopped.

Reply:
@abhinavdahiya Where do you suggest this live? In its own separate operator?

Reply:
@abhinavdahiya this resource would be used to manage the lifecycle of operators. It seems logical that it belong with the thing that manages operators. I could see an argument for OLM since no SLO has external things it would want to destroy, but either the CVO or OLM are what would make the most sense to me. 
@eparis thoughts?

Reply:
> @abhinavdahiya this resource would be used to manage the lifecycle of operators.

can you explain how? because to me this helps mange the operands of the operators and not the operators themselves. Its like operators+their operands need k8s api, and therefore CVO should manage the k8s api directly.

Some operator should handle this object and everything required around it. The operator should be able to use the CVO to create the resources for it like CRDs, empty initlal resource instance like others.

Reply:
> can you explain how?
This is a signal telling you that you are about to be destroyed. In my mind, something similar to https://developer.android.com/reference/android/app/Activity#onDestroy() or https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623111-applicationwillterminate

It seems logical that the same thing that created you will tell you that you're going away. Admittedly not a perfect analogy. It can be used by either operators or operands.

IMHO creating a separate SLO just for this signal seems overkill.

Reply:
AFAIU CVO does not mange the life cycle of operators. I agree with @abhinavdahiya with `it manges the operands of the operators and not the operators themselves`. So that's a big shift in CVO's scope if this comes to CVO.

Reply:
I am wondering if we will have status conditions, history for this custom resource. It might be useful for debugging purpose when we want to track the events that proceeded a deletion.  I think we should ask @smarterclayton about the this  as it involves CVO.

Reply:
I see [the `blockTeardown` thread][1] debating the utility of that property.  So there is some contention between:

* `blockTeardown` would be nice to have, but needs someone to own the admission hook behind it.
* CVO doesn't want to own that hook.
* Creating a new second-level operator seems heavy.

Personally, I come down with "so let's drop `blockTeardown` and suggest users scope their admin auth to not include the ability to delete `Alive` unless they assume some special role they reserve for super-dangerous things".  Then we don't need any code behind this new CRD and CR, just manifests that life somewhere that ends up in the release image for the CVO to reconcile like it does all the other CRD and CR manifests it already manages.

[1]: https://github.com/openshift/enhancements/pull/265#discussion_r407560206

Reply:
Also, if folks are wondering where the manifest would live, we have precedent in [cluster-update-keys](https://github.com/openshift/cluster-update-keys) for a Git repo with thin `OWNERS` whose sole purpose is to hold manifests that get stuffed into the release image.  Spinning up a new repo for openshift/alive or whatever seems like it would be slightly tedious from an internal-process side, but lightweight for everyone after it gets set up.